drivers/net/slip.c

```diff
@@ -30,8 +30,12 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #include <net/net_pkt.h>
 #include <net/net_if.h>
 #include <net/net_core.h>
+#include <net/dummy.h>
 #include <console/uart_pipe.h>
+
+#if defined(CONFIG_NET_L2_ETHERNET)
 #include <net/ethernet.h>
```

Comment:
There shoud be no ill effects if we include ethernet.h always.

Reply:
Ok, will change this. I made this change because I noticed that other files included it conditionally as well, e.g. https://github.com/zephyrproject-rtos/zephyr/blob/4e5c093e6674e14ef1f76bb59bdae6cbef1184d5/subsys/net/ip/net_shell.c#L41-L43